internal/app/mistapiconnector/mistapiconnector_app.go

```diff
@@ -505,38 +543,65 @@ func (mc *mac) triggerUserNew(w http.ResponseWriter, r *http.Request, lines []st
 		w.Write([]byte("false"))
 		return false
 	}
-
+	
 	for _, userWebhook := range userWebhooks {
-		if userWebhook.Url == "" {
-			glog.Errorf("User webhook %s has no URL", userWebhook.ID)
-			continue
-		}
-		glog.V(model.DEBUG).Infof("Calling playback.user.new webhook %s", userWebhook.Url)
+		
+			if userWebhook.Url == "" {
+				glog.Errorf("User webhook %s has no URL", userWebhook.ID)
+				return true
+			}
 
-		resp, err := http.Post(userWebhook.Url, "text/plain", strings.NewReader(rawRequest))
-		if err != nil {
-			glog.Errorf("Error calling playback.user.new webhook %s err=%v", userWebhook.Url, err)
-			w.WriteHeader(http.StatusOK)
-			w.Write([]byte("false"))
-			return false
-		}
-		defer resp.Body.Close()
-		body, err := ioutil.ReadAll(resp.Body)
-		if err != nil {
-			glog.Errorf("Error reading response body from playback.user.new webhook %s err=%v", userWebhook.Url, err)
-			w.WriteHeader(http.StatusOK)
-			w.Write([]byte("false"))
-			return false
-		}
+			req := &http.Request{
+				Method: "POST",
+				Header: http.Header{
+					"Content-Type": []string{"application/json"},
+				},
+				Body: ioutil.NopCloser(strings.NewReader(payload)),
+			}
 
-		if resp.StatusCode/100 != 2 {
-			glog.Errorf("Error calling playback.user.new webhook %s status code=%d body=%s", userWebhook.Url, resp.StatusCode, string(body))
-			w.WriteHeader(http.StatusOK)
-			w.Write([]byte("false"))
-			return false
-		}
+			req.URL, err = url.Parse(userWebhook.Url)
+
+			if err != nil {
+				glog.Errorf("Error parsing URL %s for user webhook %s err=%v", userWebhook.Url, userWebhook.ID, err)
+				return false
+			}
+
+			if userWebhook.SharedSecret != "" {
+				signature := mc.sign(userWebhook.SharedSecret)
+				timestamp := strconv.FormatInt(time.Now().Unix(), 10)
+				signature_header := fmt.Sprintf("t=%s,v1=%s", timestamp, signature)
+				req.Header.Add("Livepeer-Signature", signature_header)
+			}
+
+			glog.V(model.DEBUG).Infof("Calling playback.user.new webhook %s", userWebhook.Url)
+
+			resp, err := http.DefaultClient.Do(req)
+			
+			if err != nil {
+				glog.Errorf("Error calling playback.user.new webhook %s err=%v", userWebhook.Url, err)
+				w.WriteHeader(http.StatusInternalServerError)
+				w.Write([]byte("false"))
```

Comment:
Hmm so I see we return both a status code and write true/false on the body. What do these actually mean? 

I got that the boolean we return is only used to log or not, but the other ones are still a mystery to me.